envisage/egg_basket_plugin_manager.py

```diff
@@ -153,7 +153,7 @@ def _harvest_plugins_in_eggs(self, application):
     def _handle_broken_distributions(self, errors):
         logger.error("Error loading distributions: %s", errors)
         if self.on_broken_distribution is None:
-            raise SystemError("Cannot find eggs %s" % errors)
+            raise RuntimeError("Cannot find compatible eggs %s" % errors)
```

Comment:
I'm not sure there's much value in changing the error message here. (But I agree that `RuntimeError` seems like a reasonable choice.)
```suggestion
            raise RuntimeError("Cannot find eggs %s" % errors)
```